drivers/gpio/gpio_keys_zephyr.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio_keys.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(zephyr_gpio_keys, LOG_LEVEL_DBG);
+
+#define DT_DRV_COMPAT zephyr_gpio_keys
+
+#define GPIO_KEYS_CFG_DEF(i)                                                                       \
+	{                                                                                          \
+		.spec = GPIO_DT_SPEC_GET(DT_DRV_INST(i), gpios),                                   \
+		.debounce_interval_ms = DT_INST_PROP(i, debounce_interval_ms),                     \
+	}
+
+struct gpio_keys_data {
+	const struct device *dev;
+	gpio_callback_handler_t callback;
+	struct gpio_callback cb_data;
+	struct k_work_delayable work;
+	uint32_t debounce_interval_ms;
+	int pin_state;
+	int is_stable;
```

Comment:
Can these use a smaller type? `is_stable` could be a bool, `pin_state` maybe a int8_t or enum?

Reply:
I tried this initially and it gave me this assert.  
```
ASSERTION FAIL [(cfg->port_pin_mask & (gpio_port_pins_t)(1UL << (pin))) != 0U] @ WEST_TOPDIR/zephyr/include/zephyr/drivers/gpio.h:1169
	Unsupported pin
```
Changing the DT init priority to above or below `GPIO_INIT_PRIORITY` didn't seem to fix this.